scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -0,0 +1,545 @@
+Once we have PCs, we can proceed to calculate UMAP.
+```{r}
+# Use batchelor to calculate PCA from a merged group fo SCEs
+multi_pca <- batchelor::multiBatchPCA(merged_sce,
+                                      subset.row = hv_list,
+                                      batch = merged_sce$sample,
+                                      preserve.single = TRUE)
+
+# add PCA results to merged SCE object 
+reducedDim(merged_sce, "merged_PCA") <- multi_pca[[1]]
+
+# add merged_UMAP from merged_PCA
+merged_sce <- scater::runUMAP(merged_sce,
+                              dimred = "merged_PCA",
+                              name = "merged_UMAP")
+
+merged_sce
+```
+
+
+Let's first see how these uncorrected UMAPs compare:
+
+```{r uncorrected individual UMAPs}
+# UMAPs scaled separately when calculated from individual libraries:
+scater::plotReducedDim(merged_sce,
+                       dimred = "UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.2) +
+  ggplot2::ggtitle("UMAPs separately calculated")
+```
+
+```{r uncorrected merged UMAP}
+# UMAPs scaled together when calculated from the merged SCE
+scater::plotReducedDim(merged_sce,
+                       dimred = "merged_UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.2) +
+  ggplot2::ggtitle("UMAPs calculated on merged data")
+```
+The main "clustering" (not technical clustering, just what we see visually!) in these plots are the batches. 
+A well-integrated dataset will instead show much more mixing among batches, as the batch effects will have hopefully been corrected.
+How might you expect an integrated UMAP to look?
+
+
+
+
+### Integration with `fastMNN`
+
+Finally, we're ready to integrate!
+To start, we'll use the `fastMNN` approach from the Bioconductor [`batchelor` package](http://www.bioconductor.org/packages/3.16/bioc/html/batchelor.html).
+As the name suggests, this method is pretty fast! 
+It returns a new SCE object containing integrated results:
+
+```{r}
+integrated_sce <- batchelor::fastMNN(
+  # the merged SCE object
+  merged_sce,
+  # vector indicating the batches, which we created as `$sample` when formatting SCEs
+  batch = merged_sce$sample, 
+  # Optionally, we can specify genes to use as the previously-ID'd
+  #  high variance genes
+  subset.row = hv_list
+)
+
+# Let's have a look!
+integrated_sce
+```
+
+There are couple pieces of information here of interest:
+
+- The new `reconstructed` assay represents the batch-corrected normalized expression values.
+While these are calculated and used during integration, they are **NOT** stand-alone values that you should use for other applications, e.g. differential gene expression.
```

Comment:
I think we probably want to note that this only includes the genes in the subest, unless we want to add `correct.all = TRUE`.

I also think it is probably not quite true that these are used during integration; They are the result of integration, but I unless I misunderstand, that the integration happens in PC space, and these are "back-calculated" from that.